app/binders/SearchDirection.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package binders
+
+import play.api.mvc.QueryStringBindable
+
+object SearchDirection extends Enumeration {
+  type SearchDirection = Value
+
+  val FORWARD = Value("FORWARD")
+  val BACKWARD = Value("BACKWARD")
+
+  implicit val queryStringBindable: QueryStringBindable[SearchDirection] =
+    new QueryStringBindable[SearchDirection] {
+      override def bind(
+            key: String,
+            params: Map[String, Seq[String]]
+      ): Option[Either[String, SearchDirection]] =
+        params.get(key).collect {
+          case Seq(s) =>
+            SearchDirection.values.find(_.toString == s).toRight("invalid value")
```

Comment:
could the error be a bit more specific? "invalid search direction parameter" or something like that...

Reply:
SearchDirection has now been removed as it was not used.